lib/ReactViews/Map/Navigation/Items/Compass.tsx

```diff
@@ -150,7 +152,7 @@ type IStateTypes = {
 };
 
 // the compass on map
-class Compass extends React.Component<PropTypes, IStateTypes> {
+class Compass extends React.PureComponent<PropTypes, IStateTypes> {
```

Comment:
By using a `PureComponent`, react will re-render only if the state and prop values change.

Reply:
Because it only does a shallow comparison right?

Reply:
Yes [PureComponent](https://react.dev/reference/react/PureComponent) does shallow comparison. Plain `Component` does only basic `===` equality check and is not effective when the object changes.